File: _archive/engine/engine/color.lua
Color = class()

function Color.setup()
    transparent = Color(0, 0, 0, 0)
    
    white = Color(1)
    black = Color(0)
    
    gray = Color(0.5)
    
    red = Color(1, 0, 0)
    green = Color(0, 1, 0)
    blue = Color(0, 0, 1)
end

function Color:init(r, g, b, a)
    self.r = r or 0
    self.g = g or r or 0
    self.b = b or r or 0
    self.a = a or 1
end

function Color:unpack()
    return self.r, self.g, self.b, self.a
end

File: __archive/js/cc/engine/color.js
let colors;

class Colors extends Component {
    constructor() {
        super();
        colorMode(RGB, 1);
        this.push({
            transparent: color(0, 0, 0, 0),

            white: color(1, 1, 1),
            black: color(0, 0, 0),

            gray: color(0.5, 0.5, 0.5),

            red: color(1, 0, 0),
            green: color(0, 1, 0),
            blue: color(0, 0, 1),

            yellow: color(1, 1, 0),
            purple: color(1, 0, 1),
            cyan: color(0, 1, 1),
        });

        colors = this;

        Color = {
            random: randomColor,
        };
    }

    reverse(clr) {
        return color(
            1 - clr._array[0],
            1 - clr._array[1],
            1 - clr._array[2],
            1
        );
    }
}
